airflow/providers/openlineage/utils/sqlparser.py

```diff
@@ -0,0 +1,263 @@
+        )
+        return get_table_schemas(
+            hook,
+            namespace,
+            database or database_info.database,
+            SQLParser.create_information_schema_query(tables=inputs, **database_kwargs) if inputs else None,
+            SQLParser.create_information_schema_query(tables=outputs, **database_kwargs) if outputs else None,
+        )
+
+    def generate_openlineage_metadata_from_sql(
+        self,
+        sql: list[str] | str,
+        hook: BaseHook,
+        database_info: DatabaseInfo,
+        database: str | None = None,
+    ) -> OperatorLineage:
+        """
+        Parses SQL statement(s) and generate OpenLineage metadata:
+            * input tables with schemas parsed
+            * output tables with schemas parsed
+            * run facets
+            * job facets
+
+        :param sql: a SQL statement or list of SQL statement to be parsed
+        :param hook: Airflow Hook used to connect to the database
+        :param database_info: database specific information
+        :param database: when passed it takes precedence over parsed database name
+        """
+        job_facets = {"sql": SqlJobFacet(query=SQLParser.normalize_sql(sql))}
+
+        parse_result: SqlMeta | None = self.parse(sql)
+        if not parse_result:
+            return OperatorLineage(job_facets=job_facets)
+
+        run_facets: dict = {}
+
+        if parse_result.errors:
+            run_facets["extractionError"] = ExtractionErrorRunFacet(
+                totalTasks=len(sql) if isinstance(sql, list) else 1,
+                failedTasks=len(parse_result.errors),
+                errors=[
+                    ExtractionError(
+                        errorMessage=error.message,
+                        stackTrace=None,
+                        task=error.origin_statement,
+                        taskNumber=error.index,
+                    )
+                    for error in parse_result.errors
+                ],
+            )
+
+        namespace = (
+            f"{database_info.scheme}://{database_info.authority}"
+            if database_info.authority
+            else database_info.scheme
+        )
+        inputs, outputs = self.parse_table_schemas(
+            hook=hook,
+            inputs=parse_result.in_tables,
+            outputs=parse_result.out_tables,
+            namespace=namespace,
+            database=database,
+            database_info=database_info,
+        )
+
+        return OperatorLineage(
+            inputs=inputs,
+            outputs=outputs,
+            run_facets=run_facets,
+            job_facets=job_facets,
+        )
+
+    @staticmethod
+    def normalize_sql(sql: str | Iterable[str]) -> str:
+        """Makes sure to return a semicolon-separated SQL statements"""
+        if isinstance(sql, str):
+            sql = [stmt for stmt in sql.split(";") if stmt != ""]
+        sql = [obj for stmt in sql for obj in stmt.split(";") if obj != ""]
```

Comment:
This won't work for pg/PSQL's multiline blocks:

```sql
CREATE FUNCTION somefunc() RETURNS integer AS $$
BEGIN
    ...
END;
$$ LANGUAGE plpgsql```

(Can be fixed in another PR as I'm guess this is porting from OL Airflow plugin?)

Reply:
Fixed with an optional import from `DbApiHook.split_sql_string`, otherwise runs same logic.